group-coordinator/src/test/java/org/apache/kafka/coordinator/group/runtime/MultiThreadedEventProcessorTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.runtime;
+
+import org.apache.kafka.common.utils.LogContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+
+import static org.apache.kafka.test.TestUtils.DEFAULT_MAX_WAIT_MS;
+import static org.apache.kafka.test.TestUtils.waitForCondition;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Timeout(value = 60)
+public class MultiThreadedEventProcessorTest {
+
+    private static class FutureEvent<T> implements CoordinatorEvent {
+        private final int key;
+        private final CompletableFuture<T> future;
+        private final Supplier<T> supplier;
+
+        FutureEvent(
+            int key,
+            Supplier<T> supplier
+        ) {
+            this.key = key;
+            this.future = new CompletableFuture<>();
+            this.supplier = supplier;
+        }
+
+        @Override
+        public void run() {
+            future.complete(supplier.get());
+        }
+
+        @Override
+        public void complete(Throwable ex) {
+            future.completeExceptionally(ex);
+        }
+
+        @Override
+        public Integer key() {
+            return key;
+        }
+
+        public CompletableFuture<T> future() {
+            return future;
+        }
+
+        @Override
+        public String toString() {
+            return "FutureEvent(key=" + key + ")";
+        }
+    }
+
+    @Test
+    public void testCreateAndClose() throws Exception {
+        CoordinatorEventProcessor eventProcessor = new MultiThreadedEventProcessor(
```

Comment:
Are there any conditions you want to check here? Or just ensure these operations don't throw errors?

Reply:
Nope. As simple as creating and closing it without any errors.